### DISCUSSION TITLE
[Bug]: Footer not visible on pdf

### DISCUSSION THREAD

**Discussion by coreyfreeman82** [spatie/laravel-pdf#72] [Q&A]
### What happened?

I'm trying to add a footer to all my pages but the first one. I've tried using $pdf->footerView() and $pdf->footerHtml but both seem to not show anything while I am able to find it on the page using CTRL + F. I need the footer to be on the bottom of the page it's on, but when I use position absolute all the page footers will be on the first page instead of their own. This is my blade that I load with pdf()->view('my-pdf.blade.php')->footerView('footer.blade.php')->name('name.pdf');

### How to reproduce the bug

my-pdf.blade.php
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link href="https://fonts.googleapis.com/css2?family=Kadwa:wght@700&family=Lato:ital,wght@1,300&display=swap"
          rel="stylesheet">
    <link href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap"
          rel="stylesheet">
    <script src="https://cdn.tailwindcss.com"></script>
    <title>{{$project->getTestDocumentTitle()}}</title>
</head>
<body>

@include('pdfs.frontpage', ['title' => $project->getTestDocumentTitle() ])

@pageBreak

<div style="" class="shadow bg-white p-5 m-12 rounded-lg">
    <h2 class="text-2xl pb-6">@lang('testcase.testcase_plural')</h2>

    <div>
        /// content
    </div>
</div>

/// i want a footer here with an image and text
@pageBreak

@foreach($testcases as $index => $testcase)
    <div class="shadow bg-white p-5 m-12 rounded-lg">
    /// some content
    </div>

    /// i want a footer here with an image and text
    @pageBreak
@endforeach
</body>
</html>
```

footer.blade.php
```
<style>
    .footer {
        font-size: 12px;
        margin: 10px auto;
        z-index: 999999;
        color: #000;
    }
</style>

<footer>
    <div>
        <p>Page @pageNumber of @totalPages</p>
    </div>
    <div>
        <img src="{{$logo}}" width="300" alt="">
    </div>
</footer>
```

### Package Version

1.1.2

### PHP Version

8.2

### Laravel Version

10

### Which operating systems does with happen with?

macOS

**Comment by RVP04** [comment]
either header and footer visible with very low font size, when including a image it is not at all showing. When you use alt attribute in the image it is showing as image not loaded please rectify the issues

**Reply by alfonsogarza** [reply]
@RVP04 were you able to figure out why the font size is tiny or images not loading?

**Reply by RVP04** [reply]
I don't exactly what's the problem I am inlining styles

**Comment by coreyfreeman82** [comment]
After doing some more testing I found out the reason the footer is not showing is because I gave my html a background-color. When I remove this, I can see the footer but it is hidden behind the background if I set it on the html or the body. Image is still not working even when giving an alt attribute. I also can't set a background color on the footer because it doesn't do anything.

```
<style>
    footer {
        color: red;
        font-size: 16px;
        margin: 10px 0;
        z-index: 999999;
        position: relative;
        background-color: green; //not showing
    }

    .footer {
        display: flex;
        align-items: center;
        justify-content: space-between;
        width: 100%;
        background: orange; //not showing
    }
</style>

<footer>
    <div class="footer">
        <div>
            <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/9a/Laravel.svg/1200px-Laravel.svg.png" width="300" alt="logo"> //showing the broken image icon
        </div>
        <div>
            <p>Page @pageNumber of @totalPages</p>
        </div>
    </div>
</footer>
```

**Comment by ArielMejiaDev** [comment]
@coreyfreeman82 @RVP04 For images there is a PR open to solve this I think #79
